src/components/Transactions/transactions.js

```diff
@@ -16,120 +16,202 @@ import { getTransactionsQuery } from '../../services/Transaction/transaction';
-    const [open, setOpen] = useState(false);
-    const handleOpen = () => setOpen(true)
-    const handleClose = () => setOpen(false);
-    let history = useHistory();
-    let transactionsState = UseGetEntity(getTransactionsQuery(accountId));
+  const auth = useAuth();
+  const { accountId } = useParams();
+  const [accountState, setAccountsState] = useState(undefined);
+  const [isLoading, setIsLoading] = useState(false);
+  const [open, setOpen] = useState(false);
+  const handleOpen = () => setOpen(true)
+  const handleClose = () => setOpen(false);
+  let history = useHistory();
+  const initialFiler = {
+    type: 3,
+    date: {
+      from: undefined,
+      to: undefined
+    }
+  };
+  const [filter, setFilter] = useState(initialFiler);
+  let transactionsQuery = getTransactionsQuery(accountId, filter);
+  let transactionsState = UseGetEntity(transactionsQuery, filter);
+  const transactionTypeOptions = [
+    {
+      value: 1,
+      label: 'Depósito',
+    },
+    {
+      value: 2,
+      label: 'Retiro',
+    },
+    {
+      value: 3,
+      label: 'Todas',
+    }
+  ];
 
-    useEffect(() => {
-        setIsLoading(true);
-        if(auth.data && !auth.isLoading) {
-            getAccountQuery(accountId).then(result =>
-              setAccountsState(result.data())
-            ).finally(() => setIsLoading(false));
-        }
-    }, [accountId, accountState]);
+  useEffect(() => {
+    setIsLoading(true);
+    if (auth.data && !auth.isLoading) {
+      getAccountQuery(accountId).then(result =>
+        setAccountsState(result.data())
+      ).catch(() => setAccountsState(undefined)
+      ).finally(() => setIsLoading(false)
+      )
+    }
+  }, [accountId]);
 
-    return (accountState ? <>
-        <Navbar />
-        <Box>
-            <Button className='back-button' onClick={() => history.push('/accounts')}><ArrowBackIosIcon/>Volver</Button>
-        </Box>
-        <div className="transactions">
-          <Card>
-            <CardContent sx={{ flex: '1 0 auto' }}>
+  return (accountState ? <>
+    <Navbar />
+    <Box>
+      <Button className='back-button' onClick={() => history.push('/accounts')}><ArrowBackIosIcon />Volver</Button>
+    </Box>
+    <div className="transactions">
+      <Card>
+        <CardContent sx={{ flex: '1 0 auto' }}>
+          <Box>
+            <Typography component="h1" variant="h4">
+              <strong>{accountState.name}</strong>
+            </Typography>
+            <Typography component="h1" variant="h5">
+              {currencyFormatter.format(accountState.balance)}
+            </Typography>
+            <Button
+              className='add-button'
+              type="button"
+              fullWidth
+              variant="contained"
+              color="success"
+              sx={{ mt: 3, mb: 2 }}
+              onClick={handleOpen}
+            >
+              <AddIcon />
+              Agregar transacción
+            </Button>
+            <Divider />
```

Comment:
Since the code for this component is huge, maybe you could split it. into different custom components to make your code more readable